fe/fe-core/src/main/java/com/starrocks/sql/analyzer/PrivilegeCheckerV2.java

```diff
@@ -195,5 +208,92 @@ public Void visitAlterResourceStatement(AlterResourceStmt statement, ConnectCont
+            }
+            return null;
+        }
+
+        // ---------------------------------------- Show Transaction Statement ---------------------------------------------
+        @Override
+        public Void visitShowTransactionStatement(ShowTransactionStmt statement, ConnectContext context) {
+            // No authentication required
+            return null;
+        }
+
+        // ---------------------------------------- Show tablet Statement -------------------------------------------------
+        @Override
+        public Void visitShowTabletStatement(ShowTabletStmt statement, ConnectContext context) {
+            return checkStmtOperatePrivilege(context);
+        }
+
+        // ---------------------------------------- Admin operate Statement -------------------------------------------------
+
+        @Override
+        public Void visitAdminSetConfigStatement(AdminSetConfigStmt statement, ConnectContext context) {
+            return checkStmtOperatePrivilege(context);
+        }
+
+        @Override
+        public Void visitAdminSetReplicaStatusStatement(AdminSetReplicaStatusStmt statement, ConnectContext context) {
+            return checkStmtOperatePrivilege(context);
+        }
+
+        @Override
+        public Void visitAdminShowConfigStatement(AdminShowConfigStmt statement, ConnectContext context) {
+            return checkStmtOperatePrivilege(context);
+        }
+
+        @Override
+        public Void visitAdminShowReplicaDistributionStatement(AdminShowReplicaDistributionStmt statement,
+                                                                ConnectContext context) {
+            return checkStmtOperatePrivilege(context);
+        }
+
+        @Override
+        public Void visitAdminShowReplicaStatusStatement(AdminShowReplicaStatusStmt statement, ConnectContext context) {
+            return checkStmtOperatePrivilege(context);
+        }
+
+        @Override
+        public Void visitAdminRepairTableStatement(AdminRepairTableStmt statement, ConnectContext context) {
+            return checkStmtOperatePrivilege(context);
+        }
+
+        @Override
+        public Void visitAdminCancelRepairTableStatement(AdminCancelRepairTableStmt statement, ConnectContext context) {
+            return checkStmtOperatePrivilege(context);
+        }
+
+        @Override
+        public Void visitAdminCheckTabletsStatement(AdminCheckTabletsStmt statement, ConnectContext context) {
+            return checkStmtOperatePrivilege(context);
+        }
+
+        // ---------------------------------------- Common Privilege logic -------------------------------------------------
+        private Void checkStmtOperatePrivilege(ConnectContext context) {
+            if (!PrivilegeManager.checkSystemAction(context, PrivilegeType.SystemAction.OPERATE)) {
+                ErrorReport.reportSemanticException(ErrorCode.ERR_SPECIFIC_ACCESS_DENIED_ERROR, "OPERATE");
+            }
+            return null;
```

Comment:
what about the following statements?
* kill <id>
* alter system xx
* cancel alter system
* show compute nodes
* show proc
* show variables like
* set

Reply:
done